From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Apthorp <jeremya@chromium.org>
Date: Fri, 1 Mar 2019 11:20:29 -0800
Subject: fix backup includes for PTRACE_GET_THREAD_AREA on arm/arm64

<asm/ptrace-abi.h> doesn't exist in the arm/arm64 sysroots, and <sys/ptrace.h>
defines PTRACE_GET_THREAD_AREA as an enum, not a #define:

  /* Get the thread pointer of a process.  */
  PTRACE_GET_THREAD_AREA = 22,

This changes the include trigger to check for PT_GET_THREAD_AREA as well as
PTRACE_GET_THREAD_AREA to catch the arm case.

Change-Id: I9044d9f315cbb15011f397a06d03d4bec7396985

diff --git a/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc b/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
index 55394a791a54a2650fa2e1ed4d4e293b940d8fa2..805b13ba24b3f5d6b013b9f473e2e74afc29683d 100644
--- a/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
+++ b/sandbox/linux/seccomp-bpf-helpers/syscall_parameters_restrictions.cc
@@ -34,12 +34,12 @@
 #if !defined(OS_NACL_NONSFI)
 #include <sys/ioctl.h>
 #include <sys/ptrace.h>
-#if !defined(PTRACE_GET_THREAD_AREA) && defined(OS_LINUX) && \
-    !defined(OS_CHROMEOS)
+#if !defined(PTRACE_GET_THREAD_AREA) && !defined(__arm__) && \
+    !defined(__aarch64__) && defined(OS_LINUX) && !defined(OS_CHROMEOS)
 // Also include asm/ptrace-abi.h since ptrace.h in older libc (for instance
 // the one in Ubuntu 16.04 LTS) is missing PTRACE_GET_THREAD_AREA.
 #include <asm/ptrace-abi.h>
-#endif  // !PTRACE_GET_THREAD_AREA && OS_LINUX && !OS_CHROMEOS
+#endif
 #endif  // !OS_NACL_NONSFI
 
 #if defined(OS_ANDROID)
